src/bootstrap/format.rs

```diff
@@ -0,0 +1,64 @@
+//! Runs rustfmt on the repository.
+
+use crate::Build;
+use std::process::Command;
+use ignore::WalkBuilder;
+use std::path::Path;
+use build_helper::t;
+
+fn rustfmt(build: &Build, path: &Path, check: bool) {
+    let rustfmt_path = build.config.initial_rustfmt.as_ref().unwrap_or_else(|| {
+        eprintln!("./x.py fmt is not supported on this channel");
+        std::process::exit(1);
+    });
+
+    let mut cmd = Command::new(&rustfmt_path);
+    // avoid the submodule config paths from coming into play,
+    // we only allow a single global config for the workspace for now
+    cmd.arg("--config-path").arg(&build.src.canonicalize().unwrap());
+    cmd.arg("--unstable-features");
+    cmd.arg("--skip-children");
+    if check {
+        cmd.arg("--check");
+    }
+    cmd.arg(&path);
+    let cmd_debug = format!("{:?}", cmd);
+    let status = cmd.status().expect("executing rustfmt");
+    assert!(status.success(), "running {} successful", cmd_debug);
```

Comment:
One `rustfmt --check` failure should not prevent running other `rustfmt --check`, otherwise you only get one error at a time.

Reply:
When that bit was originally written we were running rustfmt once for the whole repository and it showed all errors. Is that no longer how it's running?

Reply:
It looks like @Mark-Simulacrum pushed commits to this PR to run `rustfmt` on individual files.

Reply:
Aha - maybe file a bug so someone can pick up a fix?

Reply:
Yeah, rustfmt on the whole repository was leading to errors in rustfmt (IIRC, it had some problems with ignoring files or so, I don't recall the specifics now).

This should be fixable though, like we do in tidy etc by keeping track of whether we errored and only stopping at the end.